test/e2e/daemon_set.go

```diff
@@ -148,6 +149,15 @@ func testDaemonSets(f *Framework) {
 		},
 	})
 	Expect(err).NotTo(HaveOccurred())
+	defer func() {
+		Logf("Check that reaper kills all daemon pods for %s", simpleDSName)
+		dsReaper, err := kubectl.ReaperFor("DaemonSet", c)
+		Expect(err).NotTo(HaveOccurred())
+		_, err = dsReaper.Stop(ns, simpleDSName, 0, nil)
+		Expect(err).NotTo(HaveOccurred())
+		err = wait.Poll(retryInterval, retryTimeout, checkRunningOnNoNodes(f, label))
+		Expect(err).NotTo(HaveOccurred(), "error waiting for daemon pod to be reaped")
+	}()
 
```

Comment:
I've intentionally removed it. For the complex daemon, I've chosen to add a [test](https://github.com/kubernetes/kubernetes/pull/13183/files#diff-c1525034f2fb4c408e1a0fbf3babe41fR223) that unschedules the daemon pod by removing the nodeSelector from the node's labels so by the time we would reap the daemon set, there's nothing to do. Plus that path is tested already. Perhaps it would be a good spot to test that the namespace_controller properly deletes daemon sets when the namespace is deleted or maybe just test the delete method. WDYT?